internal/sync/sync.go

```diff
@@ -178,53 +181,40 @@ func (s *Synchronizer) Status() *types.SyncStatus {
 	}
 }
 
-func (s *Synchronizer) updateState(stateDiff *types.StateDiff) error {
-	for _, deployedContract := range stateDiff.DeployedContracts {
-		err := s.SetCode(stateDiff, deployedContract)
+func (s *Synchronizer) updateState(collectedDiff *CollectorDiff) error {
+	for _, deployedContract := range collectedDiff.stateDiff.DeployedContracts {
+		err := s.SetCode(collectedDiff, &deployedContract)
 		if err != nil {
 			return err
 		}
 	}
 
-	for contractAddress, memoryCells := range stateDiff.StorageDiff {
+	for contractAddress, memoryCells := range collectedDiff.stateDiff.StorageDiff {
 		for _, cell := range memoryCells {
 			err := s.state.SetSlot(new(felt.Felt).SetString(contractAddress), cell.Address, cell.Value)
 			if err != nil {
 				return err
 			}
 		}
 	}
-	s.logger.With("Block Number", stateDiff.BlockNumber).Debug("State updated")
+	s.logger.With("Block Number", collectedDiff.stateDiff.BlockNumber).Debug("State updated")
 	return nil
 }
 
-func (s *Synchronizer) SetCode(stateDiff *types.StateDiff, deployedContract types.DeployedContract) error {
-	// Get Full Contract
-	contractFromApi, err := s.feeder.GetFullContractRaw(deployedContract.Address.Hex0x(), "",
-		strconv.FormatInt(stateDiff.BlockNumber, 10))
-	if err != nil {
-		s.logger.With("Block Number", stateDiff.BlockNumber,
-			"Contract Address", deployedContract.Address.Hex0x()).
-			Error("Error getting full contract")
-		return err
-	}
-
-	contract := new(types.Contract)
-	err = contract.UnmarshalRaw(contractFromApi)
-	if err != nil {
-		s.logger.With("Block Number", stateDiff.BlockNumber,
-			"Contract Address", deployedContract.Address.Hex0x()).
-			Error("Error unmarshalling contract")
-		return err
+func (s *Synchronizer) SetCode(collectedDiff *CollectorDiff, deployedContract *types.DeployedContract) error {
+	if deployedContract == nil {
+		ErrorNotDeployedContract := errors.New("contract not deployed")
+		return ErrorNotDeployedContract
```

Comment:
Unnecessary assignment.

```suggestion
		return errors.New("contract not deployed")
```

Reply:
done